summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala

```diff
@@ -21,43 +21,62 @@ import scala.reflect.ClassTag
 case class StormTopologyBuilder(options: Map[String, Options], jobId: JobId, stormDag: Dag[Storm]) {
   @transient private val logger = LoggerFactory.getLogger(classOf[StormTopologyBuilder])
 
-  def build(): StormTopology = {
-    val topologyBuilder = new TopologyBuilder
-    stormDag.nodes.foreach {
-      case summerNode: SummerNode[Storm] => scheduleSummerBolt(summerNode, topologyBuilder)
-      case flatMapNode: FlatMapNode[Storm] => scheduleFlatMapper(flatMapNode, topologyBuilder)
-      case sourceNode: SourceNode[Storm] => scheduleSpout(sourceNode, topologyBuilder)
-    }
-    topologyBuilder.createTopology()
-  }
+  def apply: StormTopology = scheduleEdges(scheduleNodes).build(jobId)
 
-  /**
-   * Set storm to tick our nodes every second to clean up finished futures
-   */
-  private def tickConfig = {
-    val boltConfig = new BacktypeStormConfig
-    boltConfig.put(BacktypeStormConfig.TOPOLOGY_TICK_TUPLE_FREQ_SECS, java.lang.Integer.valueOf(1))
-    boltConfig
-  }
+  private def scheduleNodes: Topology =
+    stormDag.nodes.foldLeft(Topology.EMPTY) {
+      case (topology, node@(summerNode: SummerNode[Storm])) =>
+        topology.withBolt(getNodeName(node), scheduleSummerBolt(summerNode))._2
+      case (topology, node@(flatMapNode: FlatMapNode[Storm])) =>
+        topology.withBolt(getNodeName(node), FlatMapBoltProvider(this, flatMapNode).apply)._2
+      case (topology, node@(sourceNode: SourceNode[Storm])) =>
+        topology.withSpout(getNodeName(node), SpoutProvider(this, node).apply)._2
+    }
 
-  private def scheduleFlatMapper(node: FlatMapNode[Storm], topologyBuilder: TopologyBuilder) = {
-    val bolt: BaseBolt[Any, Any] = FlatMapBoltProvider(this, node).apply
+  private def scheduleEdges(topologyWithNode: Topology): Topology =
+    stormDag.nodes.foldLeft(topologyWithNode) {
+      case (topology, node@(summerNode: SummerNode[Storm])) =>
+        val edgeType = EdgeType.AggregatedKeyValues[Any, Any](getSummerKeyValueShards(node))
+        scheduleIncomingEdges(topology, node, edgeType)
+      case (topology, node@(flatMapNode: FlatMapNode[Storm])) =>
+        val usePreferLocalDependency = getOrElse(node, DEFAULT_FM_PREFER_LOCAL_DEPENDENCY)
```

Comment:
May be refactor portion until like 48 to separate method called flatMapEdgeType?